.github/workflows/ci_bindings_d.yml

```diff
@@ -0,0 +1,71 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Bindings D CI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "core/**"
+      - "bindings/c/**"
+      - "bindings/d/**"
+      - ".github/workflows/ci_bindings_d.yml"
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        dlang: ["ldc-latest", "dmd-latest"]
```

Comment:
X86/amd64 only (WiP aarch64)
DMD is the base compiler (old proprietary Compiler)

Cross-compilation compilers:
Alternative compilations based on dmd.

LDC2 (dmd-frontend + LLVM backend)
GDC (dmd-frontend+ cc backend [already included upstream since gcc 11])

Why 2? Currently D lang version is 2 (when they included garbage collector)

Reply:
> [!NOTE]
> D2's garbage collector is not integrated into a language like Java. It is only integrated into the druntime library, and phobos2 (std) depends on druntime. However, if there are no dynamic allocations, or if you use malloc/free (explicit), then the GC (addrange/addroot) won't work.

https://dlang.org/spec/garbage.html

Reply:
`importC` like zig `@cImport/@cInclude` translate C [C11 by default] to D. And `-HC` flag translate D to C++ (why? template compat)